### PR TITLE
En 1748 improvement of page's group handling

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by joewhite101** [comment]
Retest this please

**Comment by eugeniosant** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by eugeniosant** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/portal/PageAction.java [hunk 1]
```diff
@@ -46,237 +38,122 @@
  */
 public class PageAction extends com.agiletec.apsadmin.portal.PageAction {
 
-	private static final Logger _logger = LoggerFactory.getLogger(PageAction.class);
-
-	@Override
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/portal/PageAction.java]: That is a lot of code to delete. Is this covered by a test?

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/apsadmin/portal/helper/ContentPageValidator.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        return true;
+    }
+
+    protected void checkContent(ActionSupport action, String contentId) throws ApsSystemException {
+        if (StringUtils.isNotBlank(contentId)) {
+            Content content = this.getContentManager().loadContent(contentId, true);
+            if (null == content || !content.isOnLine()) {
+                List<String> args = new ArrayList<>();
+                args.add(null == content ? contentId : content.getDescription());
+                action.addFieldError("extraGroups", action.getText("error.page.setOnlineContent.ref.offline", args));
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/apsadmin/portal/helper/ContentPageValidator.java]: log debug

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/apsadmin/portal/helper/ContentPageValidator.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        if (contentGroups.contains(Group.FREE_GROUP_NAME)) {
+            return;
+        }
+        /*
+        containsAll(Collection<?> coll1, Collection<?> coll2)
+        Returns true iff all elements of coll2 are also contained in coll1.
+         */
+        boolean check = CollectionUtils.containsAll(contentGroups, pageGroups);
+        if (!check) {
+            action.addFieldError("extraGroups", action.getText("error.page.publishedContents.incompatibleGroups", new String[]{content.getId(), content.getDescription(), contentGroups.toString()}));
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/apsadmin/portal/helper/ContentPageValidator.java]: log debug

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/apsadmin/portal/helper/ContentPageValidator.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                    continue;
+                }
+                Set<String> contentGroups = this.getContentGroups(content);
+                /*
+                containsAll(Collection<?> coll1, Collection<?> coll2)
+                Returns true iff all elements of coll2 are also contained in coll1.
+                 */
+                boolean check = CollectionUtils.containsAll(pageGroups, contentGroups);
+                if (!check) {
+                    action.addFieldError("extraGroups", action.getText("error.page.contentRef.incompatibleGroups", new String[]{contentId, content.getDescription(), contentGroups.toString()}));
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/apsadmin/portal/helper/ContentPageValidator.java]: log debug

File: plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/apsadmin/portal/helper/ContentPageValidator.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ *
+ * @author E.Santoboni
+ */
+public class ContentPageValidator implements IExternalPageValidator, ApplicationContextAware {
```
Comment (joewhite101) [inline plugins/entando-plugin-jacms/src/main/java/org/entando/entando/plugins/jacms/apsadmin/portal/helper/ContentPageValidator.java]: This class needs unit tests.

**Pull request merged**